File: app/src/main/java/com/vsa/filmoteca/data/usecase/CommentsUseCase.java
package com.vsa.filmoteca.data.usecase;

import com.twitter.sdk.android.core.Session;
import com.twitter.sdk.android.core.TwitterSession;
import com.twitter.sdk.android.core.models.Search;
import com.twitter.sdk.android.core.models.Tweet;
import com.twitter.sdk.android.core.models.User;
import com.vsa.filmoteca.data.repository.TwitterDataRepository;

import javax.inject.Inject;

import rx.Observable;

/**
 * Created by albertovecinasanchez on 14/3/16.
 */
public class CommentsUseCase {

    private TwitterDataRepository mRepository;

    @Inject
    public CommentsUseCase(TwitterDataRepository repository) {
        this.mRepository = repository;
    }

    public Session getActiveSession() {
        return mRepository.getActiveSession();
    }

    public void closeActiveSession() {
        mRepository.closeActiveSession();
    }

    public Observable<Search> tweets(TwitterSession session, String hashTag) {
        return mRepository.tweets(session, hashTag);
    }

    public Observable<Tweet> update(Session session, String hashTag, String message) {
        return mRepository.update(session, hashTag, message);
    }

    public Observable<User> verifyCredentials(Session session) {
        return mRepository.verifyCredentials(session);
    }

}

File: app/src/main/java/com/vsa/filmoteca/data/usecase/CommentsUseCase.kt
package com.vsa.filmoteca.data.usecase

import com.twitter.sdk.android.core.Session
import com.twitter.sdk.android.core.TwitterSession
import com.twitter.sdk.android.core.models.Search
import com.twitter.sdk.android.core.models.Tweet
import com.twitter.sdk.android.core.models.User
import com.vsa.filmoteca.data.repository.TwitterDataRepository
import rx.Observable
import javax.inject.Inject

/**
 * Created by albertovecinasanchez on 14/3/16.
 */
class CommentsUseCase @Inject
constructor(private val mRepository: TwitterDataRepository) {

    val activeSession: Session<*>
        get() = mRepository.activeSession

    fun closeActiveSession() {
        mRepository.closeActiveSession()
    }

    fun tweets(session: TwitterSession, hashTag: String): Observable<Search> {
        return mRepository.tweets(session, hashTag)
    }

    fun update(session: Session<*>, hashTag: String, message: String): Observable<Tweet> {
        return mRepository.update(session, hashTag, message)
    }

    fun verifyCredentials(session: Session<*>): Observable<User> {
        return mRepository.verifyCredentials(session)
    }

}
